hedera-node/src/main/java/com/hedera/services/sigs/sourcing/SigMapPubKeyToSigBytes.java

```diff
@@ -53,15 +50,18 @@
 
 	@Override
 	public byte[] sigBytesFor(byte[] pubKey) throws KeyPrefixMismatchException {
-		List<byte[]> matchingSigs = sigMap.getSigPairList()
-				.stream()
-				.filter(sp -> beginsWith(pubKey, sp.getPubKeyPrefix().toByteArray()))
-				.map(this::sigBytesFor)
-				.collect(toList());
-		if (matchingSigs.size() > 1) {
-			throw new KeyPrefixMismatchException("Source signature map is ambiguous for given public key!");
+		byte[] sigBytes = EMPTY_SIG;
+		for (SignaturePair sigPair : sigMap.getSigPairList()) {
+			final byte[] pubKeyPrefix = sigPair.getPubKeyPrefix().toByteArray();
+			final int length = pubKeyPrefix.length;
+			if (Arrays.equals(pubKeyPrefix, 0, length, pubKey, 0, length)) {
```

Comment:
This changed the original logic! We should use `beginsWith` instead!

Reply:
I could be wrong, I think semantically, this should be equivalent to original comparison.

Reply:
@qnswirlds  `beginsWith` uses the same logic as defined in line 77. I will use the method

Reply:
I don't see a problem with using `Arrays.equals` here, they seem identical to me as well.

Reply:
I didn't look at what was in `beginsWith` when I commented. But since the function is also used somewhere else, we cannot removed it. And hence, here we should call it instead of duplicating code.

Reply:
@qnswirlds  Modified to use `beginsWith` method